### ISSUE TITLE
Allow registering custom server types? Alternatives?

### ISSUE THREAD

**Issue by emirkmo** [datacontract/datacontract-cli#416]
We want to register named custom server types. For instance `clickhouse` or `superset` or `artifactory` or `vault`. However, `Server` currently requires `type` to be one of the hard-coded existing types from the json schema.

The mechanism of registering custom exporter/importer is great. **Could we do the same but allow registering custom servers?** 

Because it has been great to document in the data contract which infra applies to the data product it represents via the `Server`. And `Servers` seems like the natural place to have the cli/data platform tooling parse the infra and make things happen on said servers. 

#### Continuously expand the list of supported servers

It would be naive to assume that the limited list of servers and functionality implemented in the CLI should also limit the list of servers for the data contract. Most platforms will have _some_ need for customizing the list of servers. 

If you would you rather see an expanding list of servers, how willing are you to have servers that do not have endpoints in the cli? Basically just existing as unimplemented but supported server types? I ask because building integrations for all the different server types would be an increasingly growing time consuming task. I presume you want to limit this to a set that you want to explicitly support and have seen from your customers. However, the contract itself then becomes hampered by this limitation.

####  Add more generic server types

Alternatively, would you be open to adding some more generic server types? I can think of:

1. Database
2. Web
3. Viz (BI/Data Viz portal/endpoint/app)
4. Catalog
5. Storage

Basically allow a custom generic server type more than `local`. (I guess it could be called `custom` too but that is really unclear during contract writing stage and for consumers of the contract).

Interested to hear your preference/thoughts.

I think registering custom (but named) server types would be our preference for ease of implementation. However, upstreaming many more server types as long as the burden of adding them is not large is also fine, or adding more generic server types. It would be nice for DCS and the CLI to allow for more Server types.

**Comment by jochenchrist** [comment]
Required changes are proposed for Data Contract Specification V1.2.0 https://github.com/datacontract/datacontract-specification/pull/111

**Comment by jschoedl** [comment]
Custom server types are supported by now in ODCS. Closing as completed.
